### PR TITLE
update typemaps

### PR DESCRIPTION
**Pull request by areese**

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: src/java/jni_helper_defines.h [hunk 1]
```diff
@@ -163,31 +301,62 @@ static void ThrowNullPointerException(JNIEnv *jenv, const char *mesg) {
...
+#ifdef JAVA_DEBUG_TYPEMAPS
+        _realLen = beforeBytes+ length + afterBytes;
+        _realPtr = calloc(_realLen, 1);
+        _first = (void*) ((char*) _realPtr + beforeBytes);
+        _last = (void*) ((char*) _first + length);
+
+        memset(_realPtr, 0xDD, _realLen);
+        memset(_realPtr, 0xBB, beforeBytes);
+        memset(_first, 0, length);
+        memset((void*) ((char*) _last), 0xAA, beforeBytes);
```
Comment (timrc-git) [inline src/java/jni_helper_defines.h]: Shouldn't this be afterBytes?

File: src/java/jni_helper_defines.h [hunk 2]
```diff
@@ -198,10 +367,33 @@ class ScopedMemory {
...
+        assert(_memory == _first);
+        if (!_skipAssert) {
+            char expectedBefore[beforeBytes];
+            char expectedAfter[afterBytes];
+
+            memset(expectedBefore, 0xBB, beforeBytes);
+            memset(expectedAfter, 0xAA, afterBytes);
+
+            assert(memcmp(_realPtr, expectedBefore, beforeBytes) == 0);
+            assert(memcmp(_last, expectedAfter, beforeBytes) == 0);
```
Comment (timrc-git) [inline src/java/jni_helper_defines.h]: afterBytes?

File: src/java/jni_helper_defines.h (hunk 1 above)
Comment (areese) [inline src/java/jni_helper_defines.h]: I think so.
This is actually copied from somewhere else, I'll have to go back and look.
It's copied back here to keep the 2 piles in sync in a very ugly way.

**Pull request merged**